pandas/tests/indexes/test_base.py

```diff
@@ -302,138 +291,138 @@ def test_index_ctor_infer_periodindex(self):
-                                    np_datetime64_compat('2011-01-02')])),
-                    Index([datetime(2011, 1, 1), datetime(2011, 1, 2)])]:
-            assert isinstance(idx, DatetimeIndex)
+        assert isinstance(idx, Index)
+        assert idx.dtype == object
 
-        for idx in [Index(np.array([np_datetime64_compat('2011-01-01'),
-                                    np_datetime64_compat('2011-01-02')]),
-                          dtype=object),
-                    Index([datetime(2011, 1, 1),
-                           datetime(2011, 1, 2)], dtype=object)]:
-            assert not isinstance(idx, DatetimeIndex)
-            assert isinstance(idx, Index)
-            assert idx.dtype == object
+    @pytest.mark.parametrize("vals", [
+        [1, 2, 3], np.array([1, 2, 3], dtype=int),
+        np.array([np_datetime64_compat('2011-01-01'),
+                  np_datetime64_compat('2011-01-02')]),
+        [datetime(2011, 1, 1), datetime(2011, 1, 2)]
+    ])
+    def test_constructor_dtypes_to_categorical(self, vals):
+        idx = Index(vals, dtype='category')
+        assert isinstance(idx, CategoricalIndex)
 
-        for idx in [Index(np.array([np.timedelta64(1, 'D'), np.timedelta64(
-                1, 'D')])), Index([timedelta(1), timedelta(1)])]:
-            assert isinstance(idx, TimedeltaIndex)
+    @pytest.mark.parametrize("cast_idx", [True, False])
+    @pytest.mark.parametrize("vals", [
+        Index(np.array([np_datetime64_compat('2011-01-01'),
+                        np_datetime64_compat('2011-01-02')])),
+        Index([datetime(2011, 1, 1), datetime(2011, 1, 2)])
 
-        for idx in [Index(np.array([np.timedelta64(1, 'D'),
-                                    np.timedelta64(1, 'D')]), dtype=object),
-                    Index([timedelta(1), timedelta(1)], dtype=object)]:
-            assert not isinstance(idx, TimedeltaIndex)
+    ])
+    def test_constructor_dtypes_to_datetime(self, cast_idx, vals):
+        if cast_idx:
+            idx = Index(vals, dtype=object)
             assert isinstance(idx, Index)
             assert idx.dtype == object
+        else:
+            idx = Index(vals)
+            assert isinstance(idx, DatetimeIndex)
 
-    def test_constructor_dtypes_datetime(self):
+    @pytest.mark.parametrize("cast_idx", [True, False])
+    @pytest.mark.parametrize("vals", [
+        np.array([np.timedelta64(1, 'D'), np.timedelta64(1, 'D')]),
+        [timedelta(1), timedelta(1)]
+    ])
+    def test_constructor_dyptes_to_timedelta(self, cast_idx, vals):
+        if cast_idx:
+            idx = Index(vals, dtype=object)
+            assert isinstance(idx, Index)
+            assert idx.dtype == object
+        else:
+            idx = Index(vals)
+            assert isinstance(idx, TimedeltaIndex)
 
-        for tz in [None, 'UTC', 'US/Eastern', 'Asia/Tokyo']:
-            idx = pd.date_range('2011-01-01', periods=5, tz=tz)
-            dtype = idx.dtype
+    @pytest.mark.parametrize("tz", [
```

Comment:
move to pandas/tests/indexes/datetimes/test_timezones.py (might be a duplicate as well)